community/kernel/src/main/java/org/neo4j/kernel/impl/proc/ProcedureHolder.java

```diff
@@ -20,28 +20,30 @@
 package org.neo4j.kernel.impl.proc;
 
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.NoSuchElementException;
 
 import org.neo4j.internal.kernel.api.procs.QualifiedName;
-import org.neo4j.internal.kernel.api.procs.Signature;
 
 /**
  * Simple in memory store for procedures.
  *
  * Should only be accessed from a single thread
  * @param <T> the type to be stored
  */
-class ProcedureHolder<T extends Signature>
+class ProcedureHolder<T>
 {
     private final Map<QualifiedName,Integer> nameToId = new HashMap<>();
+    private final Map<QualifiedName,Integer> caseInsensitveName2Id = new HashMap<>();
```

Comment:
It feels a little bit wasteful to have an extra hashmap here, it feels like it should be possible without it. But since it is rarely used in the hot path we can maybe leave that as a future improvement.